File: output/copilot/python/results/q24-distinct-subsequences.py
# https://leetcode.com/problems/distinct-subsequences/

class Solution(object):
    def numDistinct(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: int
        """ 
        s_len = len(s)
        t_len = len(t)
        dp = [[0]*(s_len+1) for _ in range(t_len+1)]
        for i in range(s_len+1):
            dp[0][i] = 1
        for i in range(1, t_len+1):
            for j in range(1, s_len+1):
                dp[i][j] = dp[i][j-1]
                if s[j-1] == t[i-1]:
                    dp[i][j] += dp[i-1][j-1]
        return dp[-1][-1]
    

File: output/copilot/javascript/results/q24-distinct-subsequences.js
/** https://leetcode.com/problems/distinct-subsequences/ */

/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var numDistinct = function(s, t) {
    const dp = [];
    for (let i = 0; i <= s.length; i++) {
        dp[i] = [];
        for (let j = 0; j <= t.length; j++) {
        if (i === 0) {
            dp[i][j] = j;
        } else if (j === 0) {
            dp[i][j] = 1;
        } else {
            dp[i][j] = dp[i - 1][j];
            if (s[i - 1] === t[j - 1]) {
            dp[i][j] += dp[i - 1][j - 1];
            }
        }
        }
    }
    return dp[s.length][t.length];   
};